### DISCUSSION TITLE
No cities or region in GoAccess

### DISCUSSION THREAD

**Discussion by nesalkiN** [ZoeyVid/NPMplus#1204] [Q&A] [answered]
Hello.

Do Goaccess use the geoip-dbs?

I have added them but GoAcess does always show "N/A City, N/A Region".
If I add the GOACLA-variable with "--geoip-database /data/etc/goaccess/geoip/GeoLite2-City.mmdb" it works.

I'm on Unraid and this is the path and location inside the container:
/data/etc/goaccess/geoip/
GeoLite2-ASN.mmdb
GeoLite2-City.mmdb
GeoLite2-Country.mmdb

Do they have to be in /opt/npm/etc/goaccess/geoip as mentioned in the compose.yaml?
That path does not exist. /opt inside the container is empty. All persistent data is in a volume mounted to /data

**Comment by Zoey2936** [comment]
The documentation assumes that /data is mounted to /opt/npm

**Reply by nesalkiN** [reply]
In Unraids case, using the template, it is /mnt/user/appdata/NPMplus mounted to /data inside the container and it works. The files for proxy and all the other files that should be persistent is there and used. /opt is all empty.

**Reply by Zoey2936** [reply]
Yes if you use diffrent paths /opt is empty, the paths inside the conainer should be theese: https://github.com/ZoeyVid/NPMplus/blob/e157391c8a153adb7aa740522c85e24f4e6e7751/rootfs/usr/local/bin/start.sh#L270

**Reply by nesalkiN** [reply]
Heres mine

```
npmplus:/data/etc/goaccess/geoip# ls -l
total 55927
-rw-r--r-- 1 root root  8692359 Oct 30 09:30 GeoLite2-ASN.mmdb
-rw-r--r-- 1 root root 60845522 Oct 29 21:15 GeoLite2-City.mmdb
-rw-r--r-- 1 root root  7799496 Oct 29 21:15 GeoLite2-Country.mmdb
```

It should have rights to read them too.

**Reply by Zoey2936** [reply]
and you compose yaml?

**Reply by nesalkiN** [reply]
> and you compose yaml?

No compose on Unraid:

But this is the run command:
```
docker run
  -d
  --name='NPMplus'
  --net='eth0'
  --ip='192.168.1.30'
  --pids-limit 2048
  -e HOST_OS="Unraid"
  -e HOST_HOSTNAME="Server"
  -e HOST_CONTAINERNAME="NPMplus"
  -e 'TZ'='Europe/Stockholm'
  -e 'GOA'='true'
  -e 'NGINX_DISABLE_PROXY_BUFFERING'='true'
  -e 'DISABLE_HTTP'='true'
  -e 'NGINX_LOG_NOT_FOUND'='true'
  -e 'LOGROTATE'='true'
  -e 'DISABLE_IPV6'='true'
  -e 'ACME_EMAIL'='hello@example.com'
  -e 'LOGROTATIONS'='7'
  -e 'GOA_DISABLE_IPV6'='true'
  -e 'FULLCLEAN'='false'
  -l net.unraid.docker.managed=dockerman
  -l net.unraid.docker.webui='https://[IP]:[PORT:81]/'
  -l net.unraid.docker.icon='https://raw.githubusercontent.com/ZoeyVid/NPMplus/develop/frontend/app-images/logo-256.png'
  -v '/mnt/user/appdata/npmplus/data':'/data':'rw'
  -v '/mnt/user/appdata/npmplus/www':'/var/www':'rw'
  -v '/mnt/user/appdata/geolite2/':'/data/etc/goaccess/geoip':'ro'
  --hostname npmplus
  --restart unless-stopped 'zoeyvid/npmplus'
92e3ed0510d5da1166f1f528283a8b3c5e2414585aa71e6dcea958a6c3751c8a
```

**Reply by Zoey2936** [reply]
and goaccess doesn't load the databases at them moment without custzom cli flag?

**Reply by nesalkiN** [reply]
> and goaccess doesn't load the databases at them moment without custzom cli flag?

Yes. Addning  -e 'GOACLA'='--geoip-database /data/etc/goaccess/geoip/GeoLite2-City.mmdb' and City/Region works.

**Reply by Zoey2936** [reply]
`if [ -s /data/etc/goaccess/geoip/GeoLite2-Country.mmdb ] && [ -s /data/etc/goaccess/geoip/GeoLite2-City.mmdb ] && [ -s /data/etc/goaccess/geoip/GeoLite2-ASN.mmdb ] && echo "$GOACLA" | grep -vq "geoip-database"; then echo h; fi` can you try to run this inside thze container? it should output h

**Reply by nesalkiN** [reply]
> if [ -s /data/etc/goaccess/geoip/GeoLite2-Country.mmdb ] && [ -s /data/etc/goaccess/geoip/GeoLite2-City.mmdb ] && [ -s /data/etc/goaccess/geoip/GeoLite2-ASN.mmdb ] && echo "$GOACLA" | grep -vq "geoip-database"; then echo h; fi

No h.
Checked the paths. The files are there.
```
npmplus:/app# if [ -s /data/etc/goaccess/geoip/GeoLite2-Country.mmdb ] && [ -s /data/etc/goaccess/geoip/GeoLite2-City.mmdb ] && [ -s /data/etc/goaccess/geoip/GeoLite2-ASN.mmdb ] && echo "$GOACLA" | grep -vq "geoip-database"; then echo h; fi
npmplus:/app# 
```

**Reply by Zoey2936** [reply]
```
if [ -s /data/etc/goaccess/geoip/GeoLite2-Country.mmdb ]; then echo 1; fi
if [ -s /data/etc/goaccess/geoip/GeoLite2-City.mmdb ]; then echo 2; fi
if [ -s /data/etc/goaccess/geoip/GeoLite2-ASN.mmdb ]; then echo 3; fi
if echo "$GOACLA" | grep -vq "geoip-database"; then echo 4; fi
```
please run this and send the output

**Reply by nesalkiN** [reply]
> ```
> if [ -s /data/etc/goaccess/geoip/GeoLite2-Country.mmdb ]; then echo 1; fi
> if [ -s /data/etc/goaccess/geoip/GeoLite2-City.mmdb ]; then echo 2; fi
> if [ -s /data/etc/goaccess/geoip/GeoLite2-ASN.mmdb ]; then echo 3; fi
> if echo "$GOACLA" | grep -vq "geoip-database"; then echo 4; fi
> ```
> 
> please run this and send the output

```
1
2
3
#4 missing. :)
```

**Reply by nesalkiN** [reply]
Sorry no! I had the GOACLA still defined! Removed:

```
npmplus:/app# if [ -s /data/etc/goaccess/geoip/GeoLite2-Country.mmdb ]; then echo 1; fi
if [ -s /data/etc/goaccess/geoip/GeoLite2-City.mmdb ]; then echo 2; fi
if [ -s /data/etc/goaccess/geoip/GeoLite2-ASN.mmdb ]; then echo 3; fi
if echo "$GOACLA" | grep -vq "geoip-database"; then echo 4; fi
1
2
3
4
```

```
npmplus:/app# if [ -s /data/etc/goaccess/geoip/GeoLite2-Country.mmdb ] && [ -s /data/etc/goaccess/geoip/GeoLite2-City.mmdb ] && [ -s /data/etc/goaccess/geoip/GeoLite2-ASN.mmdb ] && echo "$GOACLA" | grep -vq "geoip-database"; then echo h; fi
h
npmplus:/app# 
```

**Reply by Zoey2936** [reply]
now it should work

**Reply by nesalkiN** [reply]
> now it should work

Nice! How? Do I need to change something?

**Reply by Zoey2936** [reply]
if NPMplus detects that the file exists and GOACLA does not contains geoip-database it add the geoip databases to the goaccess startflags, so if you open geoacess the locations should be shown now

**Reply by nesalkiN** [reply]
> if NPMplus detects that the file exists and GOACLA does not contains geoip-database it add the geoip databases to the goaccess startflags, so if you open geoacess the locations should be shown now

But we didnt change anything (or did we?) and now N/A City, N/A Region again.

**Reply by Zoey2936** [reply]
https://github.com/ZoeyVid/NPMplus/blob/e157391c8a153adb7aa740522c85e24f4e6e7751/rootfs/usr/local/bin/start.sh#L271
this line is executed if all conditions are true

**Reply by Zoey2936** [reply]
and as the condition is true, it should work

**Reply by nesalkiN** [reply]
> and as the condition is true, it should work

This is what I see reading the envs:
GOACLA=--agent-list --real-os --double-decode --anonymize-ip --anonymize-level=1 --keep-last=30 --with-output-resolver --no-query-string

**Reply by Zoey2936** [reply]
it is only changed with in the run script env

**Reply by nesalkiN** [reply]
Don't know what that means. (or maybe i understand now) 😊
It doesn't work for some reason.

**Comment by nesalkiN** [comment]
Adding this, no cities:
--geoip-database=/data/etc/goaccess/geoip/GeoLite2-Country.mmdb --geoip-database=/data/etc/goaccess/geoip/GeoLite2-City.mmdb --geoip-database=/data/etc/goaccess/geoip/GeoLite2-ASN.mmdb

If rearranging them, cities work:
--geoip-database=/data/etc/goaccess/geoip/GeoLite2-City.mmdb --geoip-database=/data/etc/goaccess/geoip/GeoLite2-Country.mmdb --geoip-database=/data/etc/goaccess/geoip/GeoLite2-ASN.mmdb

Does --geoip-database allow to be used more than once? (I think yes)

**Reply by Zoey2936** [reply]
not sure

**Reply by nesalkiN** [reply]
I think its only using the first one specified. Maybe it uses the others if the first one isn't accessible but I don't know for sure. I got the impression it would use all three for different things (city, asn, country).

**Reply by Zoey2936** [reply, marked as answer]
somehow changing the order fixes it, will be included in the next release
